cmd/influx_inspect/deletetsm/deletetsm.go

```diff
@@ -0,0 +1,161 @@
+// Package deletetsm bulk deletes a measurement from a raw tsm file.
+package deletetsm
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/influxdata/influxdb/tsdb/engine/tsm1"
+)
+
+// Command represents the program execution for "influxd deletetsm".
+type Command struct {
+	// Standard input/output, overridden for testing.
+	Stderr io.Writer
+	Stdout io.Writer
+
+	measurement string // measurement to delete
+	verbose     bool   // verbose logging
+}
+
+// NewCommand returns a new instance of Command.
+func NewCommand() *Command {
+	return &Command{
+		Stderr: os.Stderr,
+		Stdout: os.Stdout,
+	}
+}
+
+// Run executes the command.
+func (cmd *Command) Run(args ...string) (err error) {
+	fs := flag.NewFlagSet("deletetsm", flag.ExitOnError)
+	fs.StringVar(&cmd.measurement, "measurement", "", "")
+	fs.BoolVar(&cmd.verbose, "v", false, "")
+	fs.SetOutput(cmd.Stdout)
+	fs.Usage = cmd.printUsage
+	if err := fs.Parse(args); err != nil {
+		return err
+	} else if fs.NArg() == 0 {
+		fmt.Printf("path required\n\n")
+		fs.Usage()
+		return nil
+	}
+
+	if !cmd.verbose {
+		log.SetOutput(ioutil.Discard)
+	}
+
+	// Validate measurement flag.
+	if cmd.measurement == "" {
+		return fmt.Errorf("measurement name required")
+	}
+
+	// Process each TSM file.
+	for _, path := range fs.Args() {
```

Comment:
I don't think this is a blocker to merging the tool, but I think it's annoying to use if you have to provide all TSM files. It seems too easy for users to miss one out for example, unless they're familiar with tools like `xargs` and so on.

I think by default we look for tsm files (so we can use other unix tools to generate that set), but we should also provide a way for a user to provide a shard, retention policy or even a database as the argument.

Reply:
Users want to work on a subset of time ranges so it seems like there's some fine grained filters we'd need to add to work for all use cases. I think it makes sense to keep the tooling low level to allow people to filter TSM files how they want external to this command. A separate TSM finder command would make sense to me instead of baking it into each subcommand.

Reply:
SGTM. Either way, I think we could make the experience a bit better. I think a tsm filter that you can pipe into this would be nice.

Reply:
Sounds like we're thinking similarly, as I mentioned a similar approach to @rbetts regarding how I might do deletes with the `influx-tools` using a pipe filter:

```
$ influx-tools export | TSM filter | influx-tools import
```